core/src/main/java/org/elasticsearch/rest/action/search/RestSearchAction.java

```diff
@@ -74,7 +75,7 @@ public String getName() {
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
         SearchRequest searchRequest = new SearchRequest();
         request.withContentOrSourceParamParserOrNull(parser ->
-            parseSearchRequest(searchRequest, request, parser));
+            parseSearchRequest(searchRequest, request, parser, size -> searchRequest.source().size(size)));
```

Comment:
can this be `searchReqeust.source()::size`

Reply:
I read your comment and though "of course it can" and implemented it and everything failed. I then remembered why I wrote it this way in the first place: `search.source()` is null at this time.

Reply:
oh hmm but it might be null later too no? should we protect against that?

Reply:
I can't be null later but I admit to it being weird. I'll leave a comment explaining.